action/protocol/staking/protocol.go

```diff
@@ -20,6 +20,7 @@ import (
 	"github.com/iotexproject/iotex-core/action"
 	"github.com/iotexproject/iotex-core/action/protocol"
 	"github.com/iotexproject/iotex-core/pkg/log"
+	"github.com/iotexproject/iotex-core/state/factory"
```

Comment:
we may need to move CandidateNamespace to this protocol as it is only used here.